iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.inl

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_MEPOO_MEMORY_MANAGER_INL
+#define IOX_POSH_MEPOO_MEMORY_MANAGER_INL
+
+namespace iox
+{
+namespace mepoo
+{
+inline constexpr const char* asStringLiteral(const MemoryManager::Error value) noexcept
+{
+    switch (value)
+    {
+    case MemoryManager::Error::NO_MEMPOOLS_AVAILABLE:
+        return "MemoryManager::Error::NO_MEMPOOLS_AVAILABLE";
+    case MemoryManager::Error::NO_MEMPOOL_FOR_REQUESTED_CHUNK_SIZE:
+        return "MemoryManager::Error::NO_MEMPOOL_FOR_REQUESTED_CHUNK_SIZE";
+    case MemoryManager::Error::MEMPOOL_OUT_OF_CHUNKS:
+        return "MemoryManager::Error::MEMPOOL_OUT_OF_CHUNKS";
+    case MemoryManager::Error::INVALID_STATE:
+        return "MemoryManager::Error::INVALID_STATE";
+    }
+
+    return "[Undefined MemoryManager::Error]";
+}
+
+inline std::ostream& operator<<(std::ostream& stream, const MemoryManager::Error value) noexcept
+{
+    stream << asStringLiteral(value);
+    return stream;
+}
+
+inline log::LogStream& operator<<(log::LogStream& stream, const MemoryManager::Error value) noexcept
+{
+    stream << asStringLiteral(value);
+    return stream;
+}
```

Comment:
I cannot do that since `asStringLiteral` cannot be used as `constexpr` when in a `.cpp` file or at least we will get a compiler warnings which results in an error in the CI

Reply:
If an implementation does not depend on a template parameter it should be moved to the `.cpp`.

Please move the functions like this:

```
diff --git a/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.hpp b/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.hpp
index 026865823..82e3f53b5 100644
--- a/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.hpp
+++ b/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.hpp
@@ -105,13 +105,13 @@ inline constexpr const char* asStringLiteral(const MemoryManager::Error value) n
 /// @param[in] stream sink to write the message to
 /// @param[in] value to convert to a string literal
 /// @return the reference to `stream` which was provided as input parameter
-inline std::ostream& operator<<(std::ostream& stream, const MemoryManager::Error value) noexcept;
+std::ostream& operator<<(std::ostream& stream, const MemoryManager::Error value) noexcept;
 
 /// @brief Convenience stream operator to easily use the `asStringLiteral` function with iox::log::LogStream
 /// @param[in] stream sink to write the message to
 /// @param[in] value to convert to a string literal
 /// @return the reference to `stream` which was provided as input parameter
-inline log::LogStream& operator<<(log::LogStream& stream, const MemoryManager::Error value) noexcept;
+log::LogStream& operator<<(log::LogStream& stream, const MemoryManager::Error value) noexcept;
 
 } // namespace mepoo
 } // namespace iox
diff --git a/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.inl b/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.inl
index 97feb1deb..f0954e874 100644
--- a/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.inl
+++ b/iceoryx_posh/include/iceoryx_posh/internal/mepoo/memory_manager.inl
@@ -21,6 +21,7 @@ namespace iox
 {
 namespace mepoo
 {
+
 inline constexpr const char* asStringLiteral(const MemoryManager::Error value) noexcept
 {
     switch (value)
@@ -38,18 +39,6 @@ inline constexpr const char* asStringLiteral(const MemoryManager::Error value) n
     return "[Undefined MemoryManager::Error]";
 }
 
-inline std::ostream& operator<<(std::ostream& stream, const MemoryManager::Error value) noexcept
-{
-    stream << asStringLiteral(value);
-    return stream;
-}
-
-inline log::LogStream& operator<<(log::LogStream& stream, const MemoryManager::Error value) noexcept
-{
-    stream << asStringLiteral(value);
-    return stream;
-}
-
 } // namespace mepoo
 } // namespace iox
 
diff --git a/iceoryx_posh/source/mepoo/memory_manager.cpp b/iceoryx_posh/source/mepoo/memory_manager.cpp
index b555f2b62..cc748ad06 100644
--- a/iceoryx_posh/source/mepoo/memory_manager.cpp
+++ b/iceoryx_posh/source/mepoo/memory_manager.cpp
@@ -207,5 +207,17 @@ cxx::expected<SharedChunk, MemoryManager::Error> MemoryManager::getChunk(const C
         return cxx::success<SharedChunk>(SharedChunk(chunkManagement));
     }
 }
+
+std::ostream& operator<<(std::ostream& stream, const MemoryManager::Error value) noexcept
+{
+    stream << asStringLiteral(value);
+    return stream;
+}
+
+log::LogStream& operator<<(log::LogStream& stream, const MemoryManager::Error value) noexcept
+{
+    stream << asStringLiteral(value);
+    return stream;
+}
 } // namespace mepoo
 } // namespace iox
```